src/core/src/core_logic/ConfigurePatchingProcessor.py

```diff
@@ -128,7 +129,7 @@ def __report_consolidated_configure_patch_status(self, status=Constants.STATUS_T
                                                                   auto_assessment_state=self.current_auto_assessment_state)
 
     def __raise_if_agent_incompatible(self):
-        if not self.telemetry_writer.is_agent_compatible():
```

Comment:
Leave line 132 as-is, and create a exception-to-the-rule condition check above it that just return if true.

So:
if self.lifecycle_manager.get_vm_cloud_type() == Constants.VMCloudType.ARC and self.execution_config.operation not in [Constants.ASSESSMENT, Constants.Installation]:
     return